agent/version/gen/version-gen.go

```diff
@@ -72,17 +72,44 @@ func gitHash() string {
 	cmd := exec.Command("git", "rev-parse", "--short=8", "HEAD")
 	hash, err := cmd.Output()
 	if err != nil {
+		log.Printf("failed to execute git rev-parse --short=8 HEAD, err: %v", err)
 		return "UNKNOWN"
 	}
-	return strings.TrimSpace(string(hash))
+	hashStr := strings.TrimSpace(string(hash))
+	log.Printf("Successfully retrieved short hash value from git rev-parse --short=8 HEAD, hashStr: %s",
+		hashStr)
+	return hashStr
+}
+
+func releaseCommitGitHash() (string, error) {
+	fullHash, err := os.ReadFile(filepath.Join("..", "..", "RELEASE_COMMIT"))
+	if err != nil {
+		return "", fmt.Errorf("unable to read RELEASE_COMMIT file, err: %v", err)
+	}
+	fullHashStr := strings.TrimSpace(string(fullHash))
+	if fullHashStr == "" || len(fullHashStr) < 8 {
+		log.Fatalf("Invalid hash value obtained from RELEASE_COMMIT file (empty or length <8), fullHashStr: %s",
```

Comment:
nit: don't need `(empty or length <8)` since you're printing the hash anyways

Reply:
also, if you use log.Fatalf, that will do an os.Exit call, which would immediately exit the program and not fall back to reading hash using `gitHash()`: https://pkg.go.dev/log#Fatal

so may want to use Printf?

Reply:
resolved offline. exiting here is intentional and helps us fail loudly without falling back when the RELEASE_COMMIT file has invalid content.